test/e2e/e2e.go

```diff
@@ -158,6 +158,68 @@ var _ = Describe("validation", func() {
 			})
 		})
 	})
+
+	Context("MetalLB contains incorrect data", func() {
+		Context("MetalLB has incorrect name", func() {
+
+			var metallb *metallbv1alpha.Metallb
+			var metallbCRExisted bool
+
+			BeforeEach(func() {
+				metallb = &metallbv1alpha.Metallb{}
+				err := loadMetallbFromFile(metallb, MetallbCRFile)
+
+				metallb.SetName("incorrectname")
+				Expect(err).ToNot(HaveOccurred())
+
+				metallbCRExisted = true
+				err = testclient.Client.Get(context.Background(), goclient.ObjectKey{Namespace: metallb.Namespace, Name: metallb.Name}, metallb)
```

Comment:
I wonder if we should trust the tests here not to leak status instead of trying to fix leakeage

Reply:
I based this on the other tests, where we check the leak before every test.
Removing the check.